### PR TITLE
Added Eslint and Prettier configuration

### PR DESCRIPTION
**Pull request by ritiksaxena124**

### CODE REVIEW

**Comment by ritiksaxena124** [comment]
If the configurations look good then we can add the husky to check for errors before every commit.

File: .eslintrc.js [hunk 1]
```diff
@@ -15,11 +15,39 @@ module.exports = {
...
+    // For file naming in kebab-case
+    'check-file/filename-naming-convention': [
+      'error',
+      {
+        '**/*.{js,ts}': 'KEBAB_CASE',
+      },
+      { ignoreMiddleExtensions: true },
+    ],
+    // For folder naming in kebab-case
+    'check-file/folder-naming-convention': ['error', { '**/*': 'KEBAB_CASE' }],
```
Comment (abhykrss) [inline .eslintrc.js:51]: add check for 300 lines of code, excluding blank spaces and comments

File: package.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
     "start:dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
     "start:prod": "node dist/main",
-    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
+    "lint": "eslint \"{src,test}/**/*.ts\"",
+    "lint:fix": "eslint \"{src,test}/**/*.ts\" --fix",
```
Comment (abhykrss) [inline package.json:16]: Add a script called pre-commit-check  which will run prettier first then lint and if that pass, we commit.

File: .eslintrc.js [hunk 3]
```diff
@@ -15,11 +15,39 @@ module.exports = {
...
     jest: true,
   },
-  ignorePatterns: ['.eslintrc.js'],
+  ignorePatterns: ['.eslintrc.js', 'node_modules', 'dist', '.gitignore'],
   rules: {
     '@typescript-eslint/interface-name-prefix': 'off',
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/explicit-module-boundary-types': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/naming-convention': [
```
Comment (abhykrss) [inline .eslintrc.js:24]: no console logs checks

File: .eslintrc.js [hunk 4]
```diff
@@ -15,11 +15,39 @@ module.exports = {
     node: true,
     jest: true,
   },
-  ignorePatterns: ['.eslintrc.js'],
+  ignorePatterns: ['.eslintrc.js', 'node_modules', 'dist', '.gitignore'],
   rules: {
     '@typescript-eslint/interface-name-prefix': 'off',
```
Comment (abhykrss) [inline .eslintrc.js:20]: no unused vars?

File: .eslintrc.js [hunk 5]
```diff
@@ -15,11 +15,39 @@ module.exports = {
     node: true,
     jest: true,
   },
-  ignorePatterns: ['.eslintrc.js'],
+  ignorePatterns: ['.eslintrc.js', 'node_modules', 'dist', '.gitignore'],
```
Comment (abhykrss) [inline .eslintrc.js:18]: no unused vars?

Comment (abhykrss) [inline .eslintrc.js:18]: enforce indentation of 2

Comment (abhykrss) [inline .eslintrc.js:18]: enforce eqeqeq (===)

Comment (abhykrss) [inline .eslintrc.js:18]: add  a rule that says if prettier not pass, lint will fail.

File: .eslintrc.js (hunk 3 above)
Comment (abhykrss) [inline .eslintrc.js:24]: duplicate imports and semi colons required

**Review by abhykrss — CHANGES_REQUESTED** [review]
changes requested

File: .eslintrc.js (hunk 3 above)
Comment (ritiksaxena124) [reply .eslintrc.js:24]: > no console logs checks

ESLint should throw errors for every console or are we allowing `console.warn()`, `console.error()`

Comment (abhykrss) [reply .eslintrc.js:28]: console.anything not allowed if log is important, raise it using this.logger.warn("this is a warning")

File: .prettierrc [hunk 6]
```diff
@@ -1,4 +1,12 @@
...
+  "trailingComma": "es5",
   "singleQuote": true,
-  "trailingComma": "all"
+  "printWidth": 80,
+  "tabWidth": 2,
+  "useTabs": false,
+  "jsxSingleQuote": false,
+  "bracketSpacing": true,
+  "arrowParens": "avoid",
+  "endOfLine": "auto"
```
[the remark this answers is not in the document]
Comment (AjeetP-dev) [reply .prettierrc:11]: validated

**Pull request merged**